src/pages/user/institution/InstitutionSelector.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FC } from 'react';
+import styled from 'styled-components';
+import { FormControl, Select, MenuItem } from '@material-ui/core';
+import { RecursiveUnit, emptyRecursiveUnit, Subunit, FormikUnit } from '../../../types/institution.types';
+import { Field, FormikProps, useFormikContext } from 'formik';
+
+const StyledInstitutionSelector = styled.div`
+  width: 30rem;
+`;
+
+const StyledFormControl = styled(FormControl)`
+  margin-top: 1rem;
+`;
+
+interface InstitutionSelectorProps {
+  unit: RecursiveUnit;
+  counter: number;
+}
+
+const InstitutionSelector: FC<InstitutionSelectorProps> = ({ unit, counter }) => {
+  const { values, setFieldValue }: FormikProps<FormikUnit> = useFormikContext();
+
+  const handleChange = (newValue: string, previousValue: string) => {
+    const subunit = unit.subunits?.find((subunit: Subunit) => subunit.name === newValue);
+
+    if (!previousValue) {
+      setFieldValue(`subunits[${counter}].name`, newValue);
+      setFieldValue(`subunits[${counter}].id`, subunit!.id);
```

Comment:
denne åpner vel for at id kan bli satt til null
skulle vi heller kjørt en return med en gang om vi ikke finner subunit?

Reply:
will fix